File: smart-backend/src/main/java/com/bbpp/smartbackend/modules/member/entity/Member.java
package com.bbpp.smartbackend.modules.member.entity;


import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("ums_member")
public class Member {

    private Long id;

    private String username;

    private String password;

    private String nickname;

    private String phone;

    private String email;

    private String avatar;

    private Integer gender; // 1-男 2-女

    private LocalDate birthday;

    private Integer memberLevel; // 1-普通会员 2-银卡 3-金卡

    private Integer status; // 0-禁用 1-启用

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

File: smart-shop/src/api/member.ts
import request from './request'

export function login(data: { username: string; password: string }) {
    return request({
        url: '/api/member/login',
        method: 'post',
        data
    })
}

export function register(data: { username: string; password: string; nickname?: string })  {
    return request({
        url: '/api/member/register',
        method: 'post',
        data
    })
}